control_panel_api/tools.py

```diff
@@ -56,8 +74,61 @@ def deploy_for(self, user):
             '--set', f'authProxy.auth0.domain={self.auth_client_domain}',
             '--set', f'authProxy.auth0.clientId={self.auth_client_id}',
             '--set', f'authProxy.auth0.clientSecret={self.auth_client_secret}',
+        ]
+
+    def deploy_for(self, user):
+        """
+        Deploy the given tool in the user namespace.
+        >>> class RStudio(BaseTool):
+        ...     name = 'rstudio'
+        >>> rstudio = RStudio()
+        >>> rstudio.deploy_for(alice)
+        """
+
+        username = user.username.lower()
+        deploy_params = self.deploy_params(user)
+        self.helm.upgrade_release(
+            self.release_name(username),
+            self.chart_name,
+            '--namespace', user.k8s_namespace,
+            *deploy_params
         )
 
-    def _get_auth_client_config(self, key):
-        setting_key = f'{self.name}_AUTH_CLIENT_{key}'
-        return getattr(settings, setting_key.upper())
+
+class RStudio(BaseTool):
+    name = 'rstudio'
+
+
+class JupyterLab(BaseTool):
+    name = 'jupyter-lab'
+
+    def release_name(self, username):
+        return f'{self.name}-{username}'
+
+    def deploy_params(self, user):
```

Comment:
I see that this is different from RStudio params, I wonder if method in base class could return list of common ones and then we merge with tool-specific params in the subclass.

Alternatively if we override in both subclasses maybe the base class could be kind of abstract and these method would just raise an exception to tell to implement these methods.

Not that this is not work or anything, just a thought.